### DISCUSSION TITLE
Router.back() and browser back button doesnt work

### DISCUSSION THREAD

**Discussion by enzzoperez** [vercel/next.js#13518] [Help]
# Bug report

## Describe the bug
Hi all!, I try to navigate in my app but when I want to use the browser back button or `Router.back()` the url change but my app stays in the same place.

What am I doing wrong?

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Go to localhost:300
2. Click on whatever video
3. Click on Volver button or browser back button

## Expected behavior
The app should go back

## Screenshots
![gif-bug](https://user-images.githubusercontent.com/22937791/83191489-7b623500-a10a-11ea-966d-f64a5179a55a.gif)

## System information

- OS: ubuntu 19
- Browser (if applies) : chrome, firefox
- Version of Next.js: 9.3.5
- Version of Node.js: v12.13.1

## Additional context
my code for navigation is the follow

```
<Link
          href={
            props.streaming === "1"
              ? `/streaming?video=${id}`
              : `/player?video=${source_id}`
          }
>
   <a>
    {titleVideo}
   </a>
</Link>
```

and then in my player I receive the id
```
   const parsed = queryString.parse(window.location.search);
    if (parsed) {
      setVideoData(() => data.find((item) => item.id === parsed.video));
    }
```

thanks for read

**Comment by jamesmosier** [comment]
Hi @enzzoperez, did you have a repository you meant to include as a code sample?

**Reply by enzzoperez** [reply]
Hi @jamesmosier,how are you?, sorry, I have not, I can build a codesanbox.. meanwhile, I need to understand if it is required to build the statics page for each video id using `getStaticPath`, because I have a lot of ids and I dont know how that process could affect the performance

**Reply by jamesmosier** [reply]
Take a look at the [`fallback: true` option](https://nextjs.org/docs/basic-features/data-fetching#fallback-true) for `getStaticPaths`. This will allow you to prerender *some* pages but you won't need to do all of them. Instead `getStaticProps` will be called at render time for those you do not prerender

**Reply by enzzoperez** [reply]
Perfect,  I'll read more about that. Thank you for response!

**Comment by nxtman123** [comment]
We had this sort of bug in our app more than once. We found it to be caused by other code (not Links) that was touching the browser history. In one case it was via `replace` from `useRouter()`, but it was incorrectly formatted. In another, it was code that was changing the browser history directly via `window.history`.

I hope this helps anyone running into this sort of bug know where to look.

**Reply by benjaminPlaKayakomat** [reply]
thanks a lot @nxtman123, it helps a lot!

**Comment by AspireOne** [comment]
I'm having the issue right now, and no code is touching the browser page history directly...
